bigquery_etl/view/__init__.py

```diff
@@ -192,10 +192,17 @@ def _valid_view_naming(self):
         ]
         is_view_statement = (
             " ".join(tokens[0].normalized.split()) == "CREATE OR REPLACE"
-            and tokens[1].normalized == "VIEW"
+            and (
+                tokens[1].normalized == "VIEW"
+                or (
+                    tokens[1].normalized.upper() == "MATERIALIZED"
+                    and tokens[2].normalized == "VIEW"
+                )
+            )
         )
         if is_view_statement:
-            target_view = str(tokens[2]).strip().split()[0]
+            view_id_token = tokens[2] if tokens[1].normalized == "VIEW" else tokens[3]
```

Comment:
sqlparse doesn't currently recognize `MATERIALIZED` as a keyword, so `normalized` doesn't automatically uppercase it.

Reply:
Might be worth opening a bug for this. We've opened a couple in the past (like [this](https://github.com/andialbrecht/sqlparse/issues/583) one). Usually, the response/fix time is super fast

Reply:
Good idea: https://github.com/andialbrecht/sqlparse/issues/752